mlir/include/mlir/Dialect/MIOpen/Generator/Conv2dGenerator.h

```diff
@@ -65,6 +54,25 @@ class Conv2dGenerator {
                    });
     return permutation;
   }
+
+public:
+  std::string arch;
+  int num_cu = 0;
+  bool xdlops = false;
+  std::string operation = "conv2d";
+  std::string dataTypeStr = "f32";
+  int dilationHeight = 1, dilationWidth = 1;
+  int strideHeight = 1, strideWidth = 1;
+  int paddingHeight = 0, paddingWidth = 0;
+  std::string filterLayout = "kcyx";
+  std::string inputLayout = "nchw";
+  std::string outputLayout = "nkhw";
+
+  std::string kernelName;
+
+  SmallVector<int64_t, 4> filterDimension;
+  SmallVector<int64_t, 4> inputDimension;
+  SmallVector<int64_t, 4> outputDimension;
```

Comment:
Honestly speaking I don't like all those member variables being public + there are too many of them. It make it hard to track who changed what.

I'd prefer:
 - Private member variables
 - Public member function that return its populated values
 - A constructor that is able to construct a `Conv2dGenerator` from the result of public member functions

Reply:
The generator needs all these settings for genConvModule. They are either stored internally or passed in by the caller.

I don't like them public either, just didn't want to clutter up the file with accessors.. That is appropriate however, I will add them.

Reply:
You are right I don't like tons of accessor too.

It just look like to me we should change the class naming such that:
 - In a information population class, it is able to populate a lightweight structure `Conv2dInfo`
 - Then in this `Conv2dGenerator` class, it is able to accept a `Conv2dInfo` and do `genConvModule`